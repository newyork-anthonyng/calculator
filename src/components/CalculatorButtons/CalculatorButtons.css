.Calculator_Buttons_Row {
	display: flex;
}
