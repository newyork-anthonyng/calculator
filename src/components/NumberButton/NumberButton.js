import React, { PropTypes } from 'react';
import Button from '../Button/Button';

const NumberButton = ({ text, onClick }) => (
	<Button
		style={{ backgroundColor: '#434542' }}
		text={text}
		onClick={onClick}
	/>
);

NumberButton.propTypes = {
	text: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired
};

export default NumberButton;
