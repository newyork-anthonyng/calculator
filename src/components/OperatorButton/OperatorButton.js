import React, { PropTypes } from 'react';
import Button from '../Button/Button';

const OperatorButton = ({ text, onClick }) => (
	<Button
		style={{ backgroundColor: '#202020' }}
		text={text}
		onClick={onClick}
	/>
);

OperatorButton.propTypes = {
	text: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired
};

export default OperatorButton;
