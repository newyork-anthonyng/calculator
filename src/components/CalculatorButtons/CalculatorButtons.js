import React, { PropTypes } from 'react';
import NumberButton from '../NumberButton/NumberButton';
import OperatorButton from '../OperatorButton/OperatorButton';
import './CalculatorButtons.css';

const CalculatorButtons = ({ handleNumberPress, handleOperatorPress, handleDeletePress }) => {
	return (
		<div className="Calculator_Buttons">
			<div className="Calculator_Buttons_Row">
				<OperatorButton text="Delete" onClick={handleDeletePress} />
			</div>
			<div className="Calculator_Buttons_Row">
				<NumberButton text="7" onClick={handleNumberPress} />
				<NumberButton text="8" onClick={handleNumberPress} />
				<NumberButton text="9" onClick={handleNumberPress} />
				<OperatorButton text="/" onClick={handleOperatorPress} />
			</div>
			<div className="Calculator_Buttons_Row">
				<NumberButton text="4" onClick={handleNumberPress} />
				<NumberButton text="5" onClick={handleNumberPress} />
				<NumberButton text="6" onClick={handleNumberPress} />
				<OperatorButton text="x" onClick={handleOperatorPress} />
			</div>
			<div className="Calculator_Buttons_Row">
				<NumberButton text="1" onClick={handleNumberPress} />
				<NumberButton text="2" onClick={handleNumberPress} />
				<NumberButton text="3" onClick={handleNumberPress} />
				<OperatorButton text="-" onClick={handleOperatorPress} />
			</div>
			<div className="Calculator_Buttons_Row">
				<NumberButton text="0" onClick={handleNumberPress} />
				<OperatorButton text="=" onClick={handleOperatorPress} />
				<OperatorButton text="+" onClick={handleOperatorPress} />
			</div>
		</div>
	);
};

CalculatorButtons.propTypes = {
	handleNumberPress: PropTypes.func.isRequired,
	handleOperatorPress: PropTypes.func.isRequired,
	handleDeletePress: PropTypes.func.isRequired
};

export default CalculatorButtons;
