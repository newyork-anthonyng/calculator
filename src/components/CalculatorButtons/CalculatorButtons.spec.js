import React from 'react';
import {
	shallow,
	mount
} from 'enzyme';
import renderer from 'react-test-renderer';
import CalculatorButtons from './CalculatorButtons';
import NumberButton from '../NumberButton/NumberButton';
import OperatorButton from '../OperatorButton/OperatorButton';

describe('CalculatorButtons', () => {
	it('should render', () => {
		const component = renderer.create(
			<CalculatorButtons
				handleNumberPress={() => {}}
				handleOperatorPress={() => {}}
			/>
		);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should render number buttons', () => {
		const wrapper = shallow(
			<CalculatorButtons
				handleNumberPress={() => {}}
				handleOperatorPress={() => {}}
			/>
		);

		const numberButtons = wrapper.find(NumberButton);
		expect(numberButtons.length).toEqual(10);
	});

	it('should render operator buttons', () => {
		const wrapper = shallow(
			<CalculatorButtons
				handleNumberPress={() => {}}
				handleOperatorPress={() => {}}
			/>
		);

		const operatorButtons = wrapper.find(OperatorButton);
		expect(operatorButtons.length).toEqual(5);
	});

	it('should run callback when number button is pressed', () => {
		const cb = jest.fn();
		const wrapper = mount(
			<CalculatorButtons
				handleNumberPress={cb}
				handleOperatorPress={() => {}}
			/>
		);

		const number7Button = wrapper.find(NumberButton).at(0);
		number7Button.simulate('click');
		expect(cb.mock.calls[0][0]).toEqual('7');
	});

	it('should run callback when operator button is pressed', () => {
		const cb = jest.fn();
		const wrapper = mount(
			<CalculatorButtons
				handleNumberPress={() => {}}
				handleOperatorPress={cb}
			/>
		);

		const divideButton = wrapper.find(OperatorButton).at(0);
		divideButton.simulate('click');
		expect(cb.mock.calls[0][0]).toEqual('/');
	});
});
