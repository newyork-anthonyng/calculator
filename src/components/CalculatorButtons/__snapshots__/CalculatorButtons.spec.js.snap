exports[`CalculatorButtons should render 1`] = `
<div
  className="Calculator_Buttons">
  <div
    className="Calculator_Buttons_Row">
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#202020",
        }
      }>
      Delete
    </div>
  </div>
  <div
    className="Calculator_Buttons_Row">
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      7
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      8
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      9
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#202020",
        }
      }>
      /
    </div>
  </div>
  <div
    className="Calculator_Buttons_Row">
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      4
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      5
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      6
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#202020",
        }
      }>
      x
    </div>
  </div>
  <div
    className="Calculator_Buttons_Row">
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      1
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      2
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      3
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#202020",
        }
      }>
      -
    </div>
  </div>
  <div
    className="Calculator_Buttons_Row">
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#434542",
        }
      }>
      0
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#202020",
        }
      }>
      =
    </div>
    <div
      className="Button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#202020",
        }
      }>
      +
    </div>
  </div>
</div>
`;
